[#100205808] Remove API Worker

mtekel opened the pull request:
Api Worker only does asynchronous jobs. Other jobs are done by the API
server itself. In this commit we remove API Worker server and add the
worker template to the API server. This way API server will be able to
do all kinds of jobs. To scale, we now just need to add more API servers
instead of managing 2 separate pools of instances, which both need some
minimal amount of instances for resiliency...

keymon commented:
The commit also does something else: Forces the order of consul jobs to be the first. I suggest we mention that in the commit, so when somebody is checking the code, he can understand why this code is there.

Also, If the goal is reduce the number of VMs, I think more components can be deployed together.

Apart of that :+1:, it works.

dcarley commented:
> The commit also does something else: Forces the order of consul jobs to be the first. I suggest we mention that in the commit, so when somebody is checking the code, he can understand why this code is there.

Agree with this :+1: except for using "he" because they might not be male :memo:

mtekel commented:
Yes, consul first. This is another can of worms, as CF jobs, mainly consul_agent, don't correctly keep restarting, but instead stop after 60 tries (coincidentally, monit timeout is 60s) and get stuck. You then need to restart all these jobs manually (monit restart consul_agent). So the consul as this are right now, really should be deployed as first. The real fix however is for jobs to correctly keep restarting...

keymon commented:
Yes, I agree.

For the same reasoning, I also think it is good that this kind of info is in the commits, because this way we can troubleshoot and understand the code easier.

mtekel committed:
Remove API Worker

Api Worker only does asynchronous jobs. Other jobs are done by the API
server itself. In this commit we remove API Worker server and add the
worker template to the API server. This way API server will be able to
do all kinds of jobs. To scale, we now just need to add more API servers
instead of managing 2 separate pools of instances, which all need some
minimal amount of instances for resiliency...

mtekel committed:
Ensure consul is 1st in the template

Add consul as 1st job in the api.yml, because api.yml gets picked as
first by spiff when merging files. Consul should be 1st deployed job
because currently consul_agent job can't wait (keep restarting) for
longer than 60 seconds. It gets stuck afterwards, unless consul is
deployed and running. Also in the original cf-jobs.yml, consul is the
1st job...

mtekel commented:
I split it into 2 commits with proper comments.

keymon commented:
cool, merging
